trucks.js

```diff
@@ -503,9 +505,55 @@ var foodTrucks = [
 	}
 ];
 
-// this module should support the following methods:
 // getTrucks() - return all trucks
+function getTrucks() {
+	return foodTrucks;
+}
+
 // getTruck(name) - return the truck object matching 'name'
+function getTruck(name) {
+	return _.filter(foodTrucks, function(truck) {
+		return _.contains(truck, name);
+	});
+}
+
 // getFoodTypes() - return unique list of all associated food types (underscore has a function to help)
-// filterByDay(day) - return trucks with 'day' in schedule (use your filterByDay function from Module 3 homework)
+function getFoodTypes() {
+	return _.filter(foodTrucks, function(truck) {
+		return _.filter(truck.type);
+	});
+}
+
 // filterByFoodType(foodType) - return trucks with associated 'foodType'
+function filterByFoodTypes(foodType) {
+	return _.filter(foodTrucks, function(truck) {
+		return _.filter(truck.type, function(type) {
```

Comment:
This can be simplified by simply calling _.contains here instead of _.filter again. The second parameter in _.filter is a function that returns true or false based on passing a truth test. If true, return it in the filtered list, if not, filter it out.